src/target/riscv/riscv.c

```diff
@@ -1523,6 +1569,13 @@ const struct command_registration riscv_command_handlers[] = {
 		.usage = "",
 		.chain = riscv_exec_command_handlers
 	},
+	{
+		.name = "arm",
```

Comment:
Having the RISC-V code add ARM commands is very confusing. Since these modify the behavior of the RISC-V support, they ought to be under RISC-V.

I believe you're trying to allude to the fact that this uses the ARM semihosting protocol. You could put that in the individual command's help.

Reply:
No, the main reason is compatibility with existing tools. For example the Eclipse OpenOCD plug-in has some buttons to configure semihosting, which generate commands like `arm semihosting enable`. It is not reasonable to expect all tools to be updated. A secondary reason is the fact that the protocol used is indeed the one specified by ARM. If RISC-V will ever define its own semihosting protocol, then a command like `riscv semihosting enable` will make sense, but for now it does not.

Reply:
OK. Can you add a comment explaining that to the code?

Reply:
Yes, please check the text.